### ISSUE TITLE
Add support to null safety

### ISSUE THREAD

**Issue by bruno-garcia** [getsentry/sentry-dart#247]
Flutter is gearing up towards null-safety.

This issue is to track progress adding support to it.

- [X] Support on `sentry` #298
- [X] Support on `sentry_flutter` #337
- [ ] CI uses new Action #334
- [ ] Rename conflicting classes #250

Related to #99

The migration steps are here: https://dart.dev/null-safety/migration-guide

Important to note we should start by migrating our dependencies to the null safety version.

- [X] `uuid` support tracking: https://github.com/Daegalus/dart-uuid/issues/50
- [X] `http` support tracking: https://github.com/dart-lang/http/issues/501
- [X] `device_info` [seems to be almost there](https://github.com/flutter/plugins/pull/3324/files#diff-d2b030e7532ec572e6e20d53bf4e20d867156087f65f38efef4b8e5e16cdd296)

**Comment by marandaneto** [comment]
`package_info` not null-safety ready, but easy to get rid of.
`uuid` not null-safety ready, not hard to get rid of but it'd be a breaking change.
`http` not null-safety ready, hard to get rid of and it'd be a breaking change.

**Comment by marandaneto** [comment]
@bruno-garcia for me, this is a dup.
what's the motivation for creating a new one? see #99

I'd suggest either closing this one or the other.

**Comment by ueman** [comment]
[http](https://pub.dev/packages/http/versions/0.13.0-nullsafety.0), [uuid](https://pub.dev/packages/uuid/versions/3.0.0-nullsafety.0) and [device_info](https://pub.dev/packages/device_info/versions/2.0.0-nullsafety.2) are now migrated to nullsafety.

**Comment by marandaneto** [comment]
we need `package_info` though.
https://pub.dev/packages/package_info/versions
that's the easiest one to get rid of anyway.

**Comment by ueman** [comment]
[package_info](https://pub.dev/packages/package_info/versions/0.5.0-nullsafety/changelog) has also joined the null safety club. So the migration is now unblocked.

**Comment by marandaneto** [comment]
https://pub.dev/packages/http/versions/0.13.0 even went null-safety GA already, so we'll lose 10 pub points https://pub.dev/packages/sentry/score

**Comment by ueman** [comment]
https://github.com/getsentry/sentry-dart/issues/250 and https://github.com/getsentry/sentry-dart/pull/311 are missing for a null safe release.

**Comment by fzyzcjy** [comment]
Today null safety becomes **stable**! Is there any updates?

**Comment by ueman** [comment]
Sentry Dart needs to be rebased on the latest main branch. Sentry Flutter is yet to be migrated.

**Comment by fzyzcjy** [comment]
@ueman Thanks for the reply! Is there anything I can help?

**Comment by j574144** [comment]
Maybe this is reggard to safety null support:

I try update sentry 4.0.6 version  with http 0.13.0 version

```
Because sentry >=4.0.0-alpha.1 depends on http ^0.12.0 and <appName> depends on http ^0.13.0, sentry >=4.0.0-alpha.1 is forbidden.
```

**Comment by marandaneto** [comment]
@j574144 it is

**Comment by marandaneto** [comment]
@fzyzcjy thanks for that, if you have some spare time and would like to contribute, I've updated the description with the missing bits to finish this, here a resume:

Migrate the `sentry_flutter` package, https://github.com/getsentry/sentry-dart/tree/main/flutter;
CI uses new Action #334; (Dart and Flutter CI)
Rename conflicting classes #250;

Let us know if you'd like to start one of them, so we will be sure that we'll get this moving and that 2 people don't work in the same thing, I'm happy to assist with code review.

**Comment by fzyzcjy** [comment]
@marandaneto I will look at the first one firstly (not currently, but possibly today or tomorrow, I will let you know when I start doing it).

Shall I look at the https://github.com/getsentry/sentry-dart/tree/feat/null-safety branch? It seems that the main branch has no updates

By the way, when will the null-safe `sentry` package be published?

**Comment by marandaneto** [comment]
@fzyzcjy yeah start it from the branch.
both packages would be published together I'd say, so we assure compatibility.

**Comment by twogood** [comment]
uuid 3 with null safety is released now :)

**Comment by fzyzcjy** [comment]
@marandaneto I start working on it recently, and will push a PR soon.

**Comment by fzyzcjy** [comment]
@marandaneto PR created https://github.com/getsentry/sentry-dart/pull/337

**Comment by marandaneto** [comment]
I'm gonna cut a prerelease on Monday :)

**Comment by fzyzcjy** [comment]
@marandaneto Looking forward to it!

**Comment by marandaneto** [comment]
https://github.com/getsentry/sentry-dart/releases/tag/4.1.0-nullsafety.0

thanks once more @ueman and @fzyzcjy 

feedback is welcomed :)

I'll close this issue as the other 2 items have its own issue.

**Comment by pushpendraKh** [comment]
@marandaneto When can we expect the stable release for null safety?

**Comment by marandaneto** [comment]
@pushpendraKh releasing a new pre-release today and waiting for a bit more feedback, we don't want to break the user's code.

**Comment by marandaneto** [comment]
https://github.com/getsentry/sentry-dart/releases/tag/5.0.0
